packages/atomic/src/components/search-box-suggestions/atomic-search-box-query-suggestions/atomic-search-box-query-suggestions.tsx

```diff
@@ -17,42 +27,74 @@ export class AtomicSearchBoxQuerySuggestions {
 
   componentWillLoad() {
     try {
-      dispatchSearchBoxSuggestionsEvent(
-        ({engine, id, searchBoxController, numberOfQueries}) => {
-          const {registerQuerySuggest, fetchQuerySuggestions} =
-            loadQuerySuggestActions(engine);
-
-          (engine as SearchEngine<QuerySuggestionSection>).dispatch(
-            registerQuerySuggest({
-              id,
-              count: numberOfQueries,
-            })
-          );
-
-          return {
-            onInput: () =>
-              (engine as SearchEngine<QuerySuggestionSection>).dispatch(
-                fetchQuerySuggestions({
-                  id,
-                })
-              ),
-            renderItems: () =>
-              // TODO: limit values according to maxWithQuery/maxWithoutQuery
-              searchBoxController.state.suggestions.map((suggestion) => ({
-                content: <span innerHTML={suggestion.highlightedValue}></span>,
-                value: suggestion.rawValue,
-                onClick: () =>
-                  searchBoxController.selectSuggestion(suggestion.rawValue),
-              })),
-          };
-        },
-        this.host
-      );
+      this.initialize();
     } catch (error) {
       this.error = error as Error;
     }
   }
 
+  private initialize() {
+    dispatchSearchBoxSuggestionsEvent((bindings) => {
+      this.bindings = bindings;
+      const engine = bindings.engine as SearchEngine<QuerySuggestionSection>;
+      const {registerQuerySuggest, fetchQuerySuggestions} =
+        loadQuerySuggestActions(engine);
+
+      engine.dispatch(
+        registerQuerySuggest({
+          id: bindings.id,
+          count: bindings.numberOfQueries,
+        })
+      );
+
+      return {
+        position: Array.from(this.host.parentNode!.children).indexOf(this.host),
+        onInput: () =>
+          engine.dispatch(
+            fetchQuerySuggestions({
+              id: bindings.id,
+            })
+          ),
+        renderItems: () => this.renderItems(),
+      };
+    }, this.host);
+  }
```

Comment:
Maybe by eliminating `this.bindings` and moving initialize inside `dispatchSearchBoxSuggestionsEvent`, this logic could be divided?

E.g.:
```suggestion
  componentWillLoad() {
    try {
      dispatchSearchBoxSuggestionsEvent((bindings) => {
        return this.initialize(bindings);
      }, this.host);
    } catch (error) {
      this.error = error as Error;
    }
  }

  private registerQuerySuggest(engine: ..., id: string, count: number) {
    const {registerQuerySuggest} = loadQuerySuggestActions(engine);
    engine.dispatch(
      registerQuerySuggest({
        id,
        count: this.bindings.numberOfQueries
      })
    );
  }
  
  private fetchQuerySuggestions(engine: ..., id: string) {
    const {fetchQuerySuggestions} = loadQuerySuggestActions(engine);
    return engine.dispatch(
      fetchQuerySuggestions({
        id
      })
    );
  }

  private initialize(bindings: ...) {
    this.registerQuerySuggest(
      bindings.engine,
      bindings.id,
      bindings.numberOfQueries
    );
    
    return {
      position: Array.from(this.host.parentNode!.children).indexOf(this.host),
      onInput: () => this.fetchQuerySuggestions(bindings.engine, bindings.id),
      renderItems: () => this.renderItems(...),
    };
  }
```

Reply:
Moved initialize out to split the code a bit, but I prefer keeping a private easy to reuse bindings var rather than to pass arguments to all other methods